### PR TITLE
New league dtos

### PR DESCRIPTION
**Pull request by BlackYps**
This adds new dtos that I need for the league system in the client.

### CODE REVIEW

**Comment by Sheikah45** [comment]
Shouldn't most of these be extending abstractEntity?

**Comment by BlackYps** [comment]
I think all of them should actually

File: faf-commons-api/src/main/java/com/faforever/commons/api/dto/League.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+@Data
+@AllArgsConstructor
+@ToString(onlyExplicitlyIncluded = true, callSuper = true)
+@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = true)
+@Type("league")
+public class League extends AbstractEntity {
+  @ToString.Include
+  private String technicalName;
+  @ToString.Include
+  private String nameKey;
```
Comment (Sheikah45) [inline faf-commons-api/src/main/java/com/faforever/commons/api/dto/League.java:17]: Probably don't need the nameKey in the toString

File: faf-commons-api/src/main/java/com/faforever/commons/api/dto/LeagueSeasonDivision.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import com.github.jasminb.jsonapi.annotations.Type;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+@Data
+@ToString(onlyExplicitlyIncluded = true, callSuper = true)
+@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = true)
+@Type("leagueSeasonDivision")
+public class LeagueSeasonDivision extends AbstractEntity {
```
Comment (Sheikah45) [inline faf-commons-api/src/main/java/com/faforever/commons/api/dto/LeagueSeasonDivision.java:13]: should there be a technical name here?

File: faf-commons-api/src/main/java/com/faforever/commons/api/dto/LeagueSeasonDivisionSubdivision.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import com.github.jasminb.jsonapi.annotations.Type;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+@Data
+@ToString(onlyExplicitlyIncluded = true, callSuper = true)
+@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = true)
+@Type("leagueSeasonDivisionSubdivision")
+public class LeagueSeasonDivisionSubdivision extends AbstractEntity {
```
Comment (Sheikah45) [inline faf-commons-api/src/main/java/com/faforever/commons/api/dto/LeagueSeasonDivisionSubdivision.java:13]: technical name?

Comment (BlackYps) [reply faf-commons-api/src/main/java/com/faforever/commons/api/dto/LeagueSeasonDivisionSubdivision.java:13]: the api doesn't provide a technical name on both of these

**Review by Brutus5000 — APPROVED** [review]
In case the fields have changed in the meantime just open a new PR :slightly_smiling_face:

**Pull request merged**